python/python/res/simulator/batch_simulator.py

```diff
@@ -65,7 +67,9 @@ def callback(*args, **kwargs):
         self.ert = EnKFMain(self.res_config)
         self.control_keys = tuple(controls.keys())
         self.result_keys = tuple(results)
-        self.callback = callback
+
+        # must be weakref to avoid circular references
+        self._callback = weakref.ref(callback) if callback else None
```

Comment:
I guess this could just be `self._callback = weakref.ref(callbak)` - if you have a weak reference to `None`that will not harm anyone?

Reply:
Tried, and `TypeError: cannot create weak reference to 'NoneType' object`